fe/fe-core/src/main/java/org/apache/doris/datasource/jdbc/source/JdbcScanNode.java

```diff
@@ -138,6 +139,11 @@ private void createJdbcFilters() {
     private List<Expr> collectConjunctsToPushDown(List<Expr> conjunctsList, List<String> errors) {
         List<Expr> pushDownConjuncts = new ArrayList<>();
         for (Expr p : conjunctsList) {
+            if (ConnectContext.get() != null && !ConnectContext.get().getSessionVariable().jdbcCastExprPushdown) {
```

Comment:
If `ConnectContext.get()` is null, I think we should also disallow to push down cast expr

Reply:
And why not write this logic in `shouldPushDownConjunct()`?